### PR TITLE
unmark flag use case

### PR DESCRIPTION
**Pull request by Dylanflini**
# Summary

## References

# Self review

- [x] I have reviewed my PR changes and left comments/questions when necessary
- [ ] I have added a screenshot for the applied changes.
- [x] I have followed the style guidelines of this project.
- [x] I have added and ran the tests locally.
- [ ] I have ran the test coverage reports locally.
- [ ] I have made corresponding changes to the documentation (README and API).
- [ ] I have added the abstraction layers needed to separate responsibilities.

## Impact of these changes

This changes through `Component` impacts:

- `Component1`
- `Component2`
- `Component3`

## Evidences of expected changes

Add evidences here about each impact generated by your changes

# Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hot fix (fixes an issue in master branch)
- [ ] Docs (Documentation only changes)
- [ ] Style (Changes that do not affect the meaning of the code, formatting, missing semi-colons, etc)
- [ ] Refactor (Changes that neither fixes a bug or adds a feature)
- [ ] Performance (Changes that improve performance)
- [ ] Chore (Changes to the build process or auxiliary tools and libraries)

# Squash and merge

We are following [Angular commit messages guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines)

```
<type>(<scope>): <Jira-Issue subject>
<BLANK LINE>
<optional body>
<BLANK LINE>
<optional footer>
```

Types

```
feat: A new feature
fix: A bug fix
docs: Documentation only changes
style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
refactor: A code change that neither fixes a bug nor adds a feature
perf: A code change that improves performance
test: Adding missing or correcting existing tests
chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
```

Make sure you squash and merge your feature/bugfix branches into develop/release/hotfix branches using the suggested format.

Some examples.

```
feat: DP-1234 Edit assessment information
Frontend changes for editing an assessment:
- improve assign modal to receive an existing assessment
- integrate the reasign end point.
- couple of warnings removed
BREAKING CHANGE: the assign modal now receives 3 required parameters
```

More examples [here](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)

### CODE REVIEW

**Comment by Dylanflini** [comment]
I didn't refactor to join validations between mark and unmark use case because i think this refactor could make later.

This way we avoid conflicts between Pull Request

File: src/minesweeper/use-cases/unmarkFlag/unmarkFlag.spec.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import { IDataRepository } from '@minesweeper/domain/data.repository';
+import { BoardModel, Cell } from '@minesweeper/domain/models';
+import { dataRepository } from '@minesweeper/infrastructure/data';
+import { unMarkFlagUseCase, MarkFlagUCError, GeneralError } from './unMarkFlag';
+
+describe('unMarkFlagUseCase', () => {
+  /**
+   * Mock Data
+   */
+  const boardId = '111-222-333';
```
Comment (danielglobant) [inline src/minesweeper/use-cases/unmarkFlag/unmarkFlag.spec.ts:10]: Use the mock util y created in my PR

**Comment by danielglobant** [comment]
Even the functionality is ok i think we should wait the merge of the other PR to take advantage of its mock utils and general validations.

File: src/minesweeper/use-cases/unmarkFlag/unmarkFlag.spec.ts (hunk 1 above)
Comment (Dylanflini) [reply src/minesweeper/use-cases/unmarkFlag/unmarkFlag.spec.ts]: ready!

File: src/minesweeper/use-cases/validations.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
-import { BoardModel } from '@minesweeper/domain/models';
+import { BoardModel, IPosition } from '@minesweeper/domain/models';
 import { hasSamePosition, isNaturalNumber } from '@minesweeper/utils';
-import { IMarkFlagProps } from './markFlag/markFlag';
 
 export enum GeneralError {
   NOT_NATURAL_NUMBER = '[General Error] You only can add natural numbers (0,1,2,3,...)',
-  CELL_ALREADY_EXPOSED = "[Error] You can't add a flag / expose a cell  where there is an already exposed cell",
-  OUTSIDE_BOARD = "[Error] You can't add a flag / expose a cell outside the columns and rows of the board",
+  CELL_ALREADY_EXPOSED = '[Error] You cannot do any action where there is an already exposed cell',
```
Comment (danielglobant) [inline src/minesweeper/use-cases/validations.ts:6]: Great  fix !

File: src/minesweeper/use-cases/validations.ts [hunk 3]
```diff
@@ -20,8 +19,8 @@ export type ValidationType<PropsType, Type> = (props: PropsType, board: BoardMod
 type TValidationsProps = ValidationsList<
   'OUTSIDE_BOARD' | 'CELL_ALREADY_EXPOSED' | 'NOT_NATURAL_NUMBER'
 >;
-type TMakeValidations = ValidationType<IMarkFlagProps, TValidationsProps>;
-type IRunValidationsProps = ValidationType<IMarkFlagProps, void>;
+type TMakeValidations = ValidationType<IPosition, TValidationsProps>;
+type IRunValidationsProps = ValidationType<IPosition, void>;
```
Comment (danielglobant) [inline src/minesweeper/use-cases/validations.ts:23]: Good

**Review by danielglobant — APPROVED** [review]

**Pull request merged**